x-pack/filebeat/module/activemq/audit/test/audit.log

```diff
@@ -0,0 +1,4 @@
+INFO  | anonymous called org.apache.activemq.broker.jmx.QueueView.retryMessages[] at 27-11-2019 08:45:57,213 | qtp443290224-47
+INFO  | admin called org.apache.activemq.broker.jmx.QueueView.retryMessages[] at 27-11-2019 08:45:57,229 | qtp443290224-45
```

Comment:
Would it make sense to extract these timestamps to use them as `@timestamp` or as some [event](https://www.elastic.co/guide/en/ecs/1.1/ecs-event.html) timestamps?
It could be done as some future improvement in any case.

Reply:
Actually I was considering this, but eventually postponed because there are two different types of messages:
1. `admin called org.apache.activemq.broker.jmx.QueueView.retryMessages[] at 27-11-2019 08:45:57,229`
2. `admin requested /admin/createDestination.action [JMSDestination='test' JMSDestinationType='queue' secret='4eb0bc3e-9d7a-4256-844c-24f40fda98f1' ] from 127.0.0.1`

In the first case you can extract a date, in the other - IP address. Both are built using normal String concatenation. If anything is changed (additional spacebar, bracket), the field extraction process will fail.

Anyway, I'm not defending from extracting these fields. I'm open for a discussion :)

Reply:
Hmmm maybe use some javascript (script processor) to check if the last part of the message is timestamp? Just a thought, I agree that can be a separate PR if we decide to do it.

Reply:
Hmm, I'm curious if it's possible to cover this with only grok expressions :)...

The thing that worries me is the fact that both messages are schemaless. We may not notice that the format has changed in some minor update of the ActiveMQ.

Reply:
Yeah, let's not do it by now. As only some messages have the timestamp we may be disordering events.